elide-core/src/main/java/com/yahoo/elide/core/sort/Sorting.java

```diff
@@ -68,20 +64,34 @@ public Sorting(final Map<String, SortOrder> sortingRules) {
      * @return The valid sorting rules - validated through the entity dictionary, or empty dictionary
      * @throws InvalidValueException when sorting values are not valid for the jpa entity
      */
-    public <T> Map<String, SortOrder> getValidSortingRules(final Class<T> entityClass,
-                                                           final EntityDictionary dictionary)
+    public <T> Map<Path, SortOrder> getValidSortingRules(final Class<T> entityClass,
+                                                         final EntityDictionary dictionary)
             throws InvalidValueException {
-        hasValidSortingRules(entityClass, dictionary);
-        return replaceIdRule(dictionary.getIdFieldName(entityClass));
-    }
+        Map<Path, SortOrder> returnMap = new LinkedHashMap<>();
+        for (Map.Entry<String, SortOrder> entry : replaceIdRule(dictionary.getIdFieldName(entityClass)).entrySet()) {
+            String dotSeparatedPath = entry.getKey();
+            SortOrder order = entry.getValue();
 
-    /**
-     * Fetches the base rules, ignoring validation against an entity class.
-     *
-     * @return a map of the sorting rules to be applied
-     */
-    public Map<String, SortOrder> getSortingRules() {
-        return this.sortRules;
+            //Creating a path validates that the dot separated path is valid.
+            Path path = new Path(entityClass, dictionary, dotSeparatedPath);
+
+            //Validate that none of the relationships are to-many
+            for (Path.PathElement pathElement : path.getPathElements()) {
```

Comment:
Can we break this out to its own function for readability? i.e. `isValidSortingRule(Path)`. This could then be reused in the logic above.